// lodash-helpers/index.js
/*!
 * lodash-helpers <https://github.com/jonschlinkert/lodash-helpers>
 *
 * Copyright (c) 2014 Jon Schlinkert, contributors.
 * Licensed under the MIT license.
 */

'use strict';

var util = require('util');
var chalk = require('chalk');

function or(o) {
  return o || '';
}


exports.log = function (msg, context) {
  return console.error(chalk.gray(msg), or(context));
};

exports.inspect = function (msg, context) {
  if (typeof msg !== 'string') {
    context = msg;
    msg = '';
  }
  var obj = util.inspect(context, null, 10);
  return console.error(chalk.gray(msg), obj);
};

exports.error = function (msg, context) {
  msg = chalk.red(' [error] ') + msg;
  return console.error(msg, or(context));
};

exports.success = function (msg, context) {
  msg = chalk.green(' [success] ') + msg;
  return console.error(msg, or(context));
};

exports.warn = function (msg, context) {
  msg = chalk.yellow(' [warn] ') + msg;
  return console.warn(msg, or(context));
};

exports.info = function (msg, context) {
  msg = chalk.green(' [info] ') + msg;
  return console.info(msg, or(context));
};

exports.magenta = function (str) {
  return chalk.magenta(str);
};
